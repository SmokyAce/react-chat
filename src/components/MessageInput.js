import React from 'react';
import PropTypes from 'prop-types';
import Input from 'material-ui/Input';

class MessageInput extends React.Component {
  static propTypes = {
    disabled: PropTypes.bool.isRequired,
    sendMessage: PropTypes.func.isRequired,
  };
  state = {
    value: '',
  };

  handleValueChange = (event) => {
    this.setState({
      value: event.target.value,
    });
  };

  handleKeyPress = (event) => {
    const { value } = this.state;

    if (event.key === 'Enter' && value) {
      this.props.sendMessage(value);
      this.setState({ value: '' });
    }
  };

  render() {
    return (
      <Input
        fullWidth
        placeholder='Type your messageâ€¦'
        value={this.state.value}
        onChange={this.handleValueChange}
        onKeyPress={this.handleKeyPress}
        disabled={this.props.disabled}
      />
    );
  }
}

export default MessageInput;
