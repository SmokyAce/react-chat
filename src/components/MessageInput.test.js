/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageInput from './MessageInput';

const mockProps = {
  disabled: false,
  sendMessage: jest.fn(),
};

describe('<MessageInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageInput {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<MessageInput {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly disabled', () => {
    const tree = renderer.create(<MessageInput {...mockProps} disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
