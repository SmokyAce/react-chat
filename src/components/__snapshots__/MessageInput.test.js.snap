// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> renders correctly 1`] = `
<div
  className="MuiInput-root-47 MessageInput-root-45 MuiInput-fullWidth-54 MuiInput-multiline-53"
  onKeyPress={[Function]}
>
  <div
    className="MuiTextarea-root-61"
    style={
      Object {
        "height": 19,
      }
    }
  >
    <textarea
      aria-hidden="true"
      className="MuiTextarea-shadow-63 MuiTextarea-textarea-62"
      readOnly={true}
      rows="1"
      tabIndex={-1}
      value=""
    />
    <textarea
      aria-hidden="true"
      className="MuiTextarea-shadow-63 MuiTextarea-textarea-62"
      defaultValue={undefined}
      readOnly={true}
      rows={1}
      tabIndex={-1}
      value=""
    />
    <textarea
      aria-invalid={undefined}
      aria-required={undefined}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiTextarea-textarea-62 MuiInput-input-55 MuiInput-inputMultiline-58"
      defaultValue={undefined}
      disabled={false}
      id={undefined}
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="Type your message…"
      readOnly={undefined}
      required={undefined}
      rows={1}
      type="text"
      value=""
    />
  </div>
  <div
    className="MuiInputAdornment-root-64 MessageInput-sendIcon-46 MuiInputAdornment-positionEnd-66"
  >
    <button
      aria-label="Send message"
      className="MuiButtonBase-root-73 MuiIconButton-root-67"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-72"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-76"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-82"
      />
    </button>
  </div>
</div>
`;

exports[`<MessageInput /> renders correctly disabled 1`] = `
<div
  className="MuiInput-root-47 MessageInput-root-45 MuiInput-disabled-50 MuiInput-fullWidth-54 MuiInput-multiline-53"
  onKeyPress={[Function]}
>
  <div
    className="MuiTextarea-root-61"
    style={
      Object {
        "height": 19,
      }
    }
  >
    <textarea
      aria-hidden="true"
      className="MuiTextarea-shadow-63 MuiTextarea-textarea-62"
      readOnly={true}
      rows="1"
      tabIndex={-1}
      value=""
    />
    <textarea
      aria-hidden="true"
      className="MuiTextarea-shadow-63 MuiTextarea-textarea-62"
      defaultValue={undefined}
      readOnly={true}
      rows={1}
      tabIndex={-1}
      value=""
    />
    <textarea
      aria-invalid={undefined}
      aria-required={undefined}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiTextarea-textarea-62 MuiInput-input-55 MuiInput-inputDisabled-57 MuiInput-inputMultiline-58"
      defaultValue={undefined}
      disabled={true}
      id={undefined}
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="Type your message…"
      readOnly={undefined}
      required={undefined}
      rows={1}
      type="text"
      value=""
    />
  </div>
  <div
    className="MuiInputAdornment-root-64 MessageInput-sendIcon-46 MuiInputAdornment-positionEnd-66"
  >
    <button
      aria-label="Send message"
      className="MuiButtonBase-root-73 MuiIconButton-root-67"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-72"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-76"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-82"
      />
    </button>
  </div>
</div>
`;
